app/models/concerns/profile_tailoring.rb

```diff
@@ -12,6 +30,13 @@ def tailored_rule_ref_ids
     end).to_h
   end
 
+  def rule_group_ancestor_ref_ids
+    base = RuleGroup.where(id: added_rules.map(&:rule_group_id))
+    base.or(
+      RuleGroup.where(id: base.select(GROUP_ANCESTRY_IDS))
+    ).pluck(:ref_id)
```

Comment:
Let's sort them by precedence, same as the rules:

```suggestion
    ).order(:precedence).pluck(:ref_id)
```

Reply:
Maybe it's not important, but for parity...